controllers/component_build_controller.go

```diff
@@ -633,14 +639,20 @@ func GeneratePipelineRun(component appstudiov1alpha1.Component, bundle string, o
 		{Name: "revision", Value: tektonapi.ArrayOrString{Type: "string", StringVal: "{{revision}}"}},
 		{Name: "output-image", Value: tektonapi.ArrayOrString{Type: "string", StringVal: proposedImage}},
 	}
-	if dockerFile, err := devfile.SearchForDockerfile([]byte(component.Status.Devfile)); err == nil && dockerFile != nil {
+
+	dockerFile, err := devfile.SearchForDockerfile([]byte(component.Status.Devfile))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Tested with an invalid devfile by missing `schemaVersion`. The controller's `Reconcile` method returns earlier at the step of checking if `Status.Devfile` is empty or not. No chance to handle the error returned from this call.